hazelcast/src/main/java/com/hazelcast/mapreduce/impl/task/DefaultContext.java

```diff
@@ -102,11 +102,12 @@ public void finalizeCombiners() {
         Combiner<ValueIn, ?> combiner = combiners.get(key);
         if (combiner == null) {
             combiner = combinerFactory.newCombiner(key);
```

Comment:
private Function<KeyIn, Combiner<ValueIn, ?>> combinerFunction = new Function<KeyIn, Combiner<ValueIn, ?>>() {
        @Override
        public Combiner<ValueIn, ?> apply(KeyIn keyIn) {
            Combiner<ValueIn, ?> combiner = combinerFactory.newCombiner(keyIn);
            combiner.beginCombine();
            return combiner;
        }
    };

    public Combiner<ValueIn, ?> getOrCreateCombiner(KeyIn key) {
        return combiners.computeIfAbsent(key,combinerFunction);
    }
It will be like this.